lib/assets/javascripts/libraries/map_reduce_utils.js

```diff
@@ -135,10 +135,14 @@ root.calculate = function(record, population, denominator, numerator, exclusion,
 
     }
   }
+  
+  if(variables){
```

Comment:
It would be helpful to have a comment explaining why we're calculating variables separately.